src/ConnectContext.php

```diff
@@ -72,6 +73,23 @@ public function getMaxAttempts(): int
         return $this->maxAttempts;
     }
 
+    public function withExponentialBackoffFactor(float $factor): self
+    {
+        if ($factor <= 0) {
+            throw new \ValueError("Invalid exponential backoff factor ({$factor}), must be greater than 0");
+        }
+
+        $clone = clone $this;
+        $clone->backoffFactor = $factor;
+
+        return $clone;
+    }
+
+    public function getExponentialBackoffFactor(): float
```

Comment:
I guess `factor` isn't the correct word here, rather `base`? https://en.wikipedia.org/wiki/Exponentiation#Terminology